src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -32,13 +48,18 @@ protected override System.Windows.UIElement Child
 
         protected override void OnActivate(System.ComponentModel.CancelEventArgs e)
         {
-            if (_needsLoadOnNextActivate)
+            EnsureOptionPageCreated();
+
+            if (s_needsLoadOnNextActivate)
             {
-                EnsureOptionPageCreated();
-                pageControl.LoadSettings();
+                // Reset the option store to the current state of the options.
+                s_optionStore.SetOptions(s_optionService.GetOptions());
+                s_optionStore.SetRegisteredOptions(s_optionService.GetRegisteredOptions());
```

Comment:
Why was this necessary? The list of registered options really shouldn't be able to change, since that's coming from MEF.

Reply:
Ah, I will remove it then. I was just trying to ensure the option store was entirely reset.